library/manageiq_provider.py

```diff
@@ -402,6 +414,20 @@ def delete_provider(self, provider_name):
         else:
             return dict(task_id=None, changed=self.changed, msg="Provider {provider_name} doesn't exist".format(provider_name=provider_name))
 
+    def filter_empty_unrequired_fields(self, configs, existing_endpoints, fields):
```

Comment:
I wouldn't use `delete` on this one because it implies we are deleting some fields from the provider, which is the exact opposite of the intention of this method.
I will rename to `filter_unsupported_fields_from_config` which is clearer.